Documentation/proposals/202305-scrapeclasses.md

```diff
@@ -0,0 +1,189 @@
+
+## Scrape Classes
+
+* **Owners:**
+  * `eronwright`
+
+* **Related Tickets:**
+  * https://github.com/prometheus-operator/prometheus-operator/issues/4121
+  * https://github.com/prometheus-operator/prometheus-operator/issues/3922
+
+This proposal aims to introduce the concept of _scrape classes_, to enable users to leverage scrape configuration data 
+that is provided by the administrator. For example, to allow pod monitors and probes to safely use
+unsafe TLS configuration fields (e.g. `keyFile`). 
+
+## Why
+
+Today, there are some scenarios that are difficult to support in a safe way, notably the need to reference 
+a local TLS file in a scrape configuration. For example, to scrape a pod in an Istio mesh with strict mTLS would require that
+the scrape configuration use a TLS keyfile that is provided by the Prometheus server's Istio sidecar. 
+See [Istio documentation](https://istio.io/latest/docs/ops/integrations/prometheus/#tls-settings) for more details.
+
+Another motivation is to improve feature parity amongst the monitor resources. The `PodMonitor` and `Probe` resources
+aren't at parity with `ServiceMonitor` because the latter allows for unsafe TLS settings.
+
+A minor motivation is to decouple the monitor spec from low-level infrastructure details like TLS certificates.
```

Comment:
Given that probes have unique configuration elements, I suppose that the concept of a probe class may make sense. One could argue for introducing a `ProbeClass` and to use the term `MonitorClass` rather than `ScrapeClass`.